Task_DEV_4/Task_DEV_4/Parallelogram.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+namespace Task_DEV_4
+{
+    class Parallelogram : Square
+    {
+        public double Height { get; set; }
+        public double Angle { get; set; }
+        public double Width { get; set; }
+
+        public Parallelogram(double length, double height) : base(length)
+        {
+            Height = height;
+        }
+
+        public Parallelogram(double length, double angle, double width) : base(length)
+        {
+            Length = length;
+            Width = width;
+            Angle = angle;
+        }
+
+        public override double CalculateArea()
+        {
+            if (Height > 0 && Length > 0)
+            {
+                return Length * Height;
+            }
+            else if (Length > 0 && Width > 0 && Angle > 0)
+            {
+                return Length * Width * Math.Sin(Angle);
+            }
+            else
+            {
+                throw new Exception("Custom messege");
```

Comment:
What is Custom message? There should be message about error. What happens when something went wrong.

Reply:
It is better to use more specific exception instead of common `Exception`.

Reply:
https://www.completecsharptutorial.com/basic/complete-system-exception.php

Reply:
If you throw exception you need to process it somewhere. (try..catch)

Reply:
Done.